javatests/arcs/core/storage/driver/DatabaseDriverTest.kt

```diff
@@ -119,15 +116,18 @@ class DatabaseDriverTest {
 
   @Test
   fun close_resetsReceiver_removesClient() = runBlockingTest {
-    val mockDb = mock<Database>()
-    whenever(mockDb.addClient(any())).thenReturn(42)
-    val driver = buildDriver<CrdtEntity.Data>(mockDb)
+    val fakeDb = object : FakeDatabase() {
+      override suspend fun addClient(client: DatabaseClient): Int = 42
+      override suspend fun removeClient(identifier: Int) {
+        assertThat(identifier).isEqualTo(42)
```

Comment:
this test does not verify that removeClient is called. You need something like this
val removeWasCalled = false
val fakeDb = object : FakeDatabase() {
override suspend fun removeClient(identifier: Int) {
        assertThat(identifier).isEqualTo(42)
       removeWasCalled = true
...
assertThat(removeWasCalled).isTrue()